app/presenters/designations_presenter.rb

```diff
@@ -36,8 +37,19 @@ def designation_title(jurisdiction)
     "#{jurisdiction} designations"
   end
 
-  def designation_total(designations, jurisdiction)
-    (designations[jurisdiction] && designations[jurisdiction].count) || 0
+  def total_number_of_designations
+    all_pas = geo_entity.protected_areas_per_jurisdiction.to_a
+    all_pas.reduce(0) { |count, j| count += j["count"] }
+  end
+
+  def percent_of_total(jurisdiction)
+    total_per_jurisdiction = designation_total(jurisdiction)
+    (( total_per_jurisdiction.to_f / total_number_of_designations.to_f ) * 100).round(2)
+  end
+
+  def designation_total(jurisdiction)
+    all_pas = geo_entity.protected_areas_per_jurisdiction
```

Comment:
I was on the fence about that because I was only going to be reusing it once elsewhere, but if you say it's better from a performance perspective @variable is the way to go then!

Reply:
I was going to ask this before, but why the use of `||=` over just a basic assignment of `geo_entity.protected_areas_per_jurisdiction`? Am I understanding correctly that it'll short-circuit and just use the value of `@pas_per_jurisdiction` if called upon the second time as it's not false?

Reply:
Yeah. `||=` basically means "return what's on the left of it if not nil (or false) or assign what's on the right"
So the first time "the left bit" (our instance variable) will be empty so the variable gets assigned with the result of the method on the right. The second time you call `pas_per_jurisdiction`, the method will notice that the related instance variable is not nil so it will automatically return the already existing value instead of calling the `protected_areas_per_jurisdiction` method again. Hence why I said it will save some time as it won't run the same SQL query again